rust/lance-core/src/datatypes/field.rs

```diff
@@ -682,6 +682,12 @@ impl Field {
                 Ok(self.clone())
             }
             (DataType::Struct(_), DataType::Struct(_)) => {
+                // Blob v2 columns are special: they can have different struct layouts
+                // (logical input vs. descriptor struct). We treat blob v2 structs as opaque
+                // during schema set operations (union/subtract).
```

Comment:
I'm not sure I'd use the word "opaque" here.  Maybe "we treat blob v2 structs as primitive fields (like a binary column) during schema set operations?